File: Desktop/exo.structure-controle-fonction/exo.c/exercice-py/surf-spher.py
import math

def calculer_surface_sphere(rayon):
    """
    Calcule la surface d'une sphère (aire de la sphère)
    
    Args:
        rayon (float): Rayon de la sphère en unités
        
    Returns:
        float: Surface calculée
        
    Lève:
        TypeError: Si le rayon n'est pas un nombre
        ValueError: Si le rayon est négatif ou nul
    """
    if not isinstance(rayon, (int, float)):
        raise TypeError("Le rayon doit être un nombre")
    if rayon <= 0:
        raise ValueError("Le rayon doit être strictement positif")
    return 4 * math.pi * (rayon ** 2)

# Exemple d'utilisation avec saisie utilisateur
try:
    r = float(input("Entrez le rayon de la sphère : "))
    surface = calculer_surface_sphere(r)
    print(f"Surface de la sphère : {surface:.2f} unités²")
    
except ValueError as e:
    print(f"Erreur : {e}")
except TypeError as e:
    print(f"Erreur de type : {e}")
except Exception as e:
    print(f"Erreur inattendue : {e}")

File: Desktop/exo.structure-controle-fonction/exo.c/surf-spher.c
#include <stdio.h>
#include <math.h> // Pour M_PI et pow()

#define _USE_MATH_DEFINES // Nécessaire pour certaines implémentations

double calculer_surface_sphere(double rayon) {
    if (rayon <= 0) {
        fprintf(stderr, "Erreur : Le rayon doit être positif\n");
        return -1.0;
    }
    
}

int main() {
    double rayon;
    
    printf("Calcul de la surface d'une sphère\n");
    printf("Entrez le rayon de la sphère : ");
    
    if (scanf("%lf", &rayon) != 1) {
        fprintf(stderr, "Erreur : Saisie invalide\n");
        return 1;
    }
    
    double surface = calculer_surface_sphere(rayon);
    
    if (surface != -1.0) {
        printf("Surface de la sphère : %.2f unités²\n", surface);
        return 0;
    }
    return 1;
}